Use strict option in the majority of the core module and in the tracker module in Angular

kaidohallik opened the pull request:
This PR is next peace for the #10631

This PR resolves problems in the most places of **core module** and entirely in the **tracker module**.

I started only with `core/auth` folder, but as tracker module is related then I included also tracker module.

Unresolved stay after this PR:
* `core/login/login-modal.service.ts` - this can be done after in all places is `open()` used as `void`, account module PR #10639 is doing that already for account module
* `core/user/user.model.ts` - this can be done in conjunction with user-management module

Some notes about changes:

* there where memory leaks in tracker component and service (missing unsubscribes from observables), in my testing these are removed now

* following direction taken in #10220 I removed prefixes from tracker component and service

* UAA login returns from server object with tokens and other information, but as this object is not used then I didn't declare model object for that information and left return type to `any`

* user logout was not reflected in tracker component, because `trackerService.disconnect` was not called on user logout (`accountService.authenticate(null)` is used for logout), now this is fixed

* simplified `account.service` and removed repeated code from `account.service`

* `hasAnyAuthority` in `account.service` uses callback to check if user has authority and typescript compiler thinks that `this.userIdentity` can be changed before this callback function is called and gives null check error, currently I resolved this by defining local variable `account` and using this in callback function, so typescript compiler can be sure that no one changes this to `null` before callback function is called

* in `auth-jwt.service getToken()` for UAA was constantly returning `null` and the only usage was in tracker service where actions were performed on `if (getToken()) { .....` - so dead code and I removed that from both places (tracker is working fine with UUA in my testing, so this was really unnecessary code)

* in `auth-jwt.service` for JWT authentication the function `authenticateSuccess` was:
    * in old AngularJs style, this is now replaced with correct Typescript code
    * token was read from http header, now this is read from service response body

* in `auth-session.service` is comment "**to get a new csrf token call the api**" and under that was condition `<%_ if (applicationType !== 'gateway') { _%>`, this resulted the following behavior in 3 possible cases:
    * session authentication (only monolith is possible) is calling this and necessary work is done
    * oauth2 gateway doesn't call this and all is working fine
    * oauth2 monolith made this call but this is not necessary because in oauth authentication flow is user redirected to other page for authentication and after authentication redirected back to JHipster application, so I removed this redundant call also from oauth2 monolith and tested with both Keycloak and Okta and all is working fine

* FYI some links about this `gateway` condition in `auth-session.service`:
    * SO question: https://stackoverflow.com/questions/48660523/jhipster-with-oauth2-oidc-option-csrf-disabled-in-generated-java-config-but-ca?noredirect=1#comment84470664_48660523
    * Issue: #7090
    * PR adding gateway condition: #7129

* simplified tracker service and replaced promises with rxjs objects (mostly promises was removed by #10383, but this file was untouched by #10383)

* test "**should call /account only once**" in `accout.service.spec` was not testing what it claimed it tests, but as test "**should call /account again if user has logged out**" tested also the first one then I removed redundant test and changed correct test name to reflect all cases that it tests


-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

<!--
Please also reference the issue number in a commit message to [automatically close the related Github issue](https://help.github.com/articles/closing-issues-via-commit-messages/)

Note: It is also possible to add `[skip ci]` to your commit message to skip Travis tests
-->

wmarques commented on generators/client/templates/angular/src/main/webapp/app/core/tracker/tracker.service.ts.ejs:
```diff
@@ -112,28 +114,29 @@ export class <%=jhiPrefixCapitalized%>TrackerService {
+            return;
+        }
+      
+        this.connectionSubscription = this.connectionSubject.subscribe(() => {
```
Can we avoid nested `subscribe` by replacing the first `subscribe` with `tap` operator  ?

kaidohallik replied:
These are different subscriptions, outer is `rxjs` one and inner is `Stomp` one, so can't avoid nesting.

kaidohallik commented on generators/client/templates/angular/src/main/webapp/app/admin/tracker/tracker.component.ts.ejs:
```diff
@@ -50,14 +52,17 @@ export class <%=jhiPrefixCapitalized%>TrackerComponent implements OnInit, OnDest
-    ngOnDestroy() {
+    ngOnDestroy(): void {
         this.trackerService.unsubscribe();
+        if (this.subscription !== null) {
```
Simplification done.

kaidohallik commented:
I resolved conflicts with master and used `?:` according to https://github.com/jhipster/generator-jhipster/pull/10639#discussion_r348303798 and used correct unsubscribe order to avoid problems in possible edge cases and simplified code by using `!`.

kaidohallik commented:
I used too eagerly `!` in tracker service so rule `@typescript-eslint/no-unnecessary-type-assertion` was violated so I reverted using `!` in this PR, I amended last commit and removed using `!` (restored if checks) and force pushed changes.

wmarques commented:
Thanks alot @kaidohallik !
